hc-dna/zomes/perspective_diff_sync/src/workspace.rs

```diff
@@ -346,12 +347,13 @@ impl Workspace {
                 branches.dedup();
 
                 for branch_index in 0..branches.len() {
+                    iterations += 1;
                     println!("===Workspace.collect_until_common_ancestor(): collect_until_common_ancestor 2.1");
                     let current_hash = branches[branch_index].clone();
                     println!("Checking current hash: {:#?}", hash_to_node_id(current_hash.clone()));
 
                     let already_visited = search.found_ancestors.borrow().contains(&current_hash);
-                    let seen_on_other_side = other.found_ancestors.borrow().contains(&current_hash);
+                    let seen_on_other_side = other.found_ancestors.borrow().contains(&current_hash) || other.bfs_branches.borrow().contains(&current_hash);
```

Comment:
This doesn't seem right. If current is NOT (yet) in found_ancestors on the other side but somewhere in it's unprocessed branches, it was not seen there (yet).

Not sure about how `seen_on_other_side` gets used.. since all tests pass with this, but I would define "seen" by having been processed from `bfs_branches` over to `found_ancestors`.